Wavefront.SDK.CSharp/Common/Application/HeartbeaterService.cs

```diff
@@ -53,11 +51,32 @@ public class HeartbeaterService : IDisposable
             ApplicationTags applicationTags,
             IList<string> components,
             string source)
+            : this(wavefrontMetricSender, applicationTags, components, source, Logging.LoggerFactory)
+        { }
+
+        /// <summary>
+        /// Construct a HeartbeaterService that periodically reports heartbeats for multiple
+        /// components.
+        /// </summary>
+        /// <param name="wavefrontMetricSender">
+        /// Sender that handles sending of heartbeat metrics to Wavefront.
+        /// </param>
+        /// <param name="applicationTags">Application tags.</param>
+        /// <param name="components">List of components to send heartbeats for.</param>
+        /// <param name="source">The source (or host).</param>
+        /// <param name="source">The logger factory used to create a logger.</param>
+        public HeartbeaterService(
+            IWavefrontMetricSender wavefrontMetricSender,
+            ApplicationTags applicationTags,
+            IList<string> components,
+            string source,
+            ILoggerFactory loggerFactory)
         {
             this.wavefrontMetricSender = wavefrontMetricSender;
             this.source = source;
             heartbeatMetricTagsList = new List<IDictionary<string, string>>();
             customTagsSet = new ConcurrentDictionary<IDictionary<string, string>, bool>(new TagsDictionaryComparer());
+            logger = loggerFactory.CreateLogger<HeartbeaterService>() ?? throw new ArgumentNullException(nameof(loggerFactory));
```

Comment:
line too long
want to avoid horizontal scrolling when doing code reviews :)

Reply:
fixed